src/java_bytecode/java_bytecode_language.cpp

```diff
@@ -257,11 +258,29 @@ bool java_bytecode_languaget::typecheck(
     get_message_handler());
 
   // now typecheck all
-  if(java_bytecode_typecheck(
-       symbol_table, get_message_handler(), string_refinement_enabled))
-    return true;
+  bool res=java_bytecode_typecheck(symbol_table, get_message_handler(), string_refinement_enabled);
+  // NOTE (FOTIS): There is some unintuitive logic here, where java_bytecode_check will return
+  // TRUE if typechecking failed, and FALSE if everything went well...
+  if (res)
+  {
+    // there is no point in continuing to concretise the generic types if typechecking failed.
+    return res;
+  }
 
-  return false;
+  generate_java_generic_typet instantiate_generic_type{get_message_handler()};
+  // check out the symbols in the symbol table at this point to see if we
+  // have a a generic type in.
+  for(const auto &symbol : symbol_table.symbols)
+  {
+    // If there is one, concretise it and add it to the symbol table.
+    if(is_java_generic_type(symbol.second.type))
+    {
+      auto new_class_symbol=instantiate_generic_type(
```

Comment:
`new_class_symbol` is not used

Reply:
Since this is the only use of this functor, you might consider just not returning the symbol.